elks/arch/i86/drivers/block/bioshd.c

```diff
@@ -332,15 +332,42 @@ static int bioshd_open(struct inode *inode, struct file *filp)
     return 0;
 }
 
+void bioshd_park(struct drive_infot *drivep) {
+    unsigned int cyls = drivep->cylinders;
+    if (cyls > 1024 || drivep->fdtype != -1) return; /* int 13 AH=0xC supports only 10 bits for cyl */
+    unsigned char bios_drive = bios_drive_map[drivep - drive_info];
+    unsigned short cyl_sec = ((cyls & 0xFF) << 8) | ((cyls & 0x300) >> 2) | 0x1; /* 0x1 = sector */
+    asm (
```

Comment:
For kernel programming, we worry a lot about the stack space any given function uses (as well as chains of function sequences), so it should be as visible as possible. For this reason, we like to see all local variables declared separately at the start of a function, instead of declared when they are needed.